src/core/activity/activityChart/activityChartPanelModels/analogSignalPlotModel.ts

```diff
@@ -27,20 +51,26 @@ export class AnalogSignalPlotModel extends AnalogSignalPanelModel {
     this.updateTime();
 
     this.state.recordsVisible.forEach((record: NodeRecord) => {
-      if (record.id === 'V_m') {
-        // Update spike threshold for membrane potential.
-        this.updateSpikeThresholdLine(record);
+      if (record.id === 'V_m' && this.params[2].visible) {
+        // Add spike threshold for membrane potential.
+        this.addSpikeThresholdLine(record);
       }
 
       if (record.nodeSize === 1) {
-        // Update line for a single node.
-        this.updateSingleLine(record);
+        // Add line for a single node.
+        this.addSingleLine(record);
       } else if (record.nodeSize > 1) {
-        // Update multiple lines for population.
-        this.updateMultipleLines(record);
-        // Update average line for population.
-        this.updateAverageLine(record);
+        // Add multiple lines for the population.
+        this.addMultipleLines(record);
+
+        // Add average line for the population.
+        if (this.params[1].value) {
+          this.addAverageLine(record);
+        }
       }
+
+      // Add active line when select on neuron unit id.
```

Comment:
I do not understand this comment in detail. Could you reformulate it a bit? :slightly_smiling_face: